src/providers/mssql/qgsmssqlprovider.cpp

```diff
@@ -1934,3 +1960,324 @@ QGISEXTERN QgsVectorLayerImport::ImportError createEmptyLayer(
+                                .arg( styleName.isEmpty() ? dsUri.table() : styleName ),
+                                QMessageBox::Yes | QMessageBox::No ) == QMessageBox::No )
+      {
+        errCause = QObject::tr( "Operation aborted. No changes were made in the database" );
+        QgsDebugMsg("User selected not to overwrite styles");
+        return false;
+      }
+
+    QgsDebugMsg("Updating styles");
+    sql = QString( "UPDATE layer_styles "
+                   " SET useAsDefault=%1"
+                   ",styleQML=%2"
+                   ",styleSLD=%3"
+                   ",description=%4"
+                   ",owner=%5"
+                   " WHERE f_table_catalog=%6"
+                   " AND f_table_schema=%7"
+                   " AND f_table_name=%8"
+                   " AND f_geometry_column=%9"
+                   " AND styleName=%10" )
+          .arg( useAsDefault ? "1" : "0" )
+          .arg( QgsMssqlProvider::quotedValue( qmlStyle ) )
+          .arg( QgsMssqlProvider::quotedValue( sldStyle ) )
+          .arg( QgsMssqlProvider::quotedValue( styleDescription.isEmpty() ? QDateTime::currentDateTime().toString() : styleDescription ) )
+          .arg( QgsMssqlProvider::quotedValue( dsUri.username() ) )
+          .arg( QgsMssqlProvider::quotedValue( dsUri.database() ) )
+          .arg( QgsMssqlProvider::quotedValue( dsUri.schema() ) )
+          .arg( QgsMssqlProvider::quotedValue( dsUri.table() ) )
+          .arg( QgsMssqlProvider::quotedValue( dsUri.geometryColumn() ) )
+          .arg( QgsMssqlProvider::quotedValue( styleName.isEmpty() ? dsUri.table() : styleName ) );
+  }
+  if ( useAsDefault )
+  {
+    QString removeDefaultSql = QString( "UPDATE layer_styles "
+                                        " SET useAsDefault=0"
+                                        " WHERE f_table_catalog=%1"
+                                        " AND f_table_schema=%2"
+                                        " AND f_table_name=%3"
+                                        " AND f_geometry_column=%4" )
+                               .arg( QgsMssqlProvider::quotedValue( dsUri.database() ) )
+                               .arg( QgsMssqlProvider::quotedValue( dsUri.schema() ) )
+                               .arg( QgsMssqlProvider::quotedValue( dsUri.table() ) )
+                               .arg( QgsMssqlProvider::quotedValue( dsUri.geometryColumn() ) );
+    sql = QString( "%1; %2;" ).arg( removeDefaultSql, sql );
+  }
+
+  QgsDebugMsg("Inserting styles");
+  QgsDebugMsg(sql);
+  bool execOk = query.exec( sql );
+
+  if ( !execOk )
+  {
+    errCause = QObject::tr( "Unable to save layer style. It's not possible to insert a new record into the style table. Maybe this is due to table permissions. Please contact your database administrator." );
+  }
+  return execOk;
+}
+
+
+QGISEXTERN QString loadStyle( const QString& uri, QString& errCause )
+{
+  QString style = "";
```

Comment:
No need to initialise here, just:

```
QString style;
```

is fine, and avoids the need to parse the empty string.